### ISSUE TITLE
Achilles result concept count query unnecessary cast

### ISSUE THREAD

**Issue by alondhe** [OHDSI/WebAPI#2194]
### Expected behavior
Achilles result concept count table can be created successfully based on /ddl/achilles endpoint SQL.

### Actual behavior
Errors on casting of concept_id to varchar.

### Steps to reproduce behavior
Go to /ddl/achilles?dialect=redshift&schema=<schema>&vocabSchema=<vocabSchema>, run SQL.

I think we shouldn't cast to varchar.

**Comment by chrisknoll** [comment]
Can you reference the specific cast that is causing the issue?

**Comment by alondhe** [comment]
Sure, these: https://github.com/OHDSI/WebAPI/blob/master/src/main/resources/ddl/achilles/achilles_result_concept_count.sql#L23-L29

I'm not sure why the casting to varchar is needed. But if there is a good reason, I think the final insert should do a final cast back to int to satisfy the DDL.

**Comment by chrisknoll** [comment]
Hi, i'm working on a branch to update this logic, but the core issue is that the concept_ids are stored as varchar() in stratum_1 (and stratum_2) in the results schema  while the join is on concept_id in concept ancestor. Because of issues with casting stratum_1 to int, we instead need to cast the concept_id in the concept_ancstor table into varchar.

In the new logic, we'll only be calculating counts for those concepts that have counts in the Achilles results table or any concept that has a descendant in the results table.

**Comment by alondhe** [comment]
Forgot to add in the error:

```
java.sql.SQLException: [Amazon](500310) Invalid operation: column "concept_id" is of type integer but expression is of type character varying 
Hint: You will need to rewrite or cast the expression.;
```

At least in Redshift, I am able to remove the casts and keep them as ints, and the join to stratum strings still works. But I'm sure that's not the case in other dialects. I think as long as we cast at the end before the insert, we should be good.

**Comment by chrisknoll** [comment]
Right, I'll make sure those elements are covered in the final version for 2.13.

Just a note:  these MMP systems are wierd:   sometimes it will try to apply the join conditions first, and apply the expressions on the select second, thus the WEHRE clause restricts which casts are being done on stratum_1.

On the same environment but maybe under different caching or load conditions, it may decide it should just take all the row results (including the casts) and then filter out the rows based on the joins, resulting in an attempted cast of stratum_1 to an int when the column contains a non-int value from another analysis.   The way to get it right every time is to do the least-constraining option which is to cast the concept_id into a varchar for the join.   You are correct, the stratum_1 should be cast down to an int for an insert into the table.

**Comment by chrisknoll** [comment]
@alondhe , I believe this issue was resolved in making the changes for #2172. Would you be able to check?

**Comment by alondhe** [comment]
With 2.13, this works now, thanks!
